src/Compilers/CSharp/Test/Semantic/Semantics/DynamicTests.cs

```diff
@@ -1510,6 +1510,217 @@ public void Goo()
+                {{type}} d = 1;
+                S s = default;
+                ref S r = ref s;
+                r.M(d);
+                System.Console.Write(s.X);
+
+                struct S
+                {
+                    public int X { get; private set; }
+                    public void M(int x) => X = x;
+                }
+                """;
+            CompileAndVerify(new[] { source, UnscopedRefAttributeDefinition }, new[] { CSharpRef },
+                expectedOutput: "1").VerifyDiagnostics();
+        }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/68063")]
+        public void RefReturn_Field([CombinatorialValues("int", "dynamic")] string type)
+        {
+            var source = $$"""
+                {{type}} d = 1;
+                S s = default;
+                P p = new P(ref s);
+                p.S.M(d);
+                System.Console.Write(s.X);
+
+                ref struct P(ref S s)
+                {
+                    public ref S S = ref s;
+                }
+
+                struct S
+                {
+                    public int X { get; private set; }
+                    public void M(int x) => X = x;
+                }
+                """;
+            CompileAndVerify(new[] { source, UnscopedRefAttributeDefinition }, targetFramework: TargetFramework.Net70,
+                expectedOutput: RefFieldTests.IncludeExpectedOutput("1"), verify: Verification.FailsPEVerify).VerifyDiagnostics();
+        }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/68063")]
+        public void StructReceiver_Field([CombinatorialValues("int", "dynamic")] string type, bool ro)
+        {
+            var source = $$"""
+                new C(default).Run();
+
+                class C(S s)
+                {
+                    private {{(ro ? "readonly" : "")}} S s = s;
+
+                    public void Run()
+                    {
+                        {{type}} d = 1;
+                        s.M(d);
+                        System.Console.Write(s.X);
+                    }
+                }
+
+                struct S
+                {
+                    public int X { get; private set; }
+                    public void M(int x) => X = x;
+                }
+                """;
+            CompileAndVerify(new[] { source, UnscopedRefAttributeDefinition }, new[] { CSharpRef },
+                expectedOutput: ro ? "0" : "1", verify: ro ? Verification.FailsPEVerify : default).VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/68063")]
+        public void StructReceiver_Rvalue()
+        {
+            var source = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        dynamic d = 1;
+                        GetS().M(d);
+                        System.Console.Write(GetS().X);
```

Comment:
>System.Console.Write(GetS().X); [](http://example.com/codeflow?start=24&length=31)

It looks like this line isn't meaningful. It is guaranteed to produce the same output regardless of what the previous line is doing.